### PR TITLE
Allow control over ebs optimization of nodes

### PR DESCRIPTION
**Pull request by chadcatlett**

### CODE REVIEW

File: contrib/main.yml.sample [hunk 1]
```diff
@@ -145,6 +145,10 @@ worker_instance_count: 3
 # disk is an io1 EBS volume, not instance storage.
 worker_instance_root_disk_size_gb: 64
 
+# worker_instance_ebs_optimized controls whether or not the instance
+# is provisioned as an EBS-optimized instance. This defaults to true.
+#worker_instance_ebs_optimized: true
```
Comment (bcwaldon) [inline contrib/main.yml.sample]: Please update the description of worker_instance_type given that its stated requirement is not always true.

File: roles/configure/files/stack-template.json [hunk 2]
```diff
@@ -3338,6 +3340,11 @@
       "Type": "Number",
       "Description": "Number of IOPs to provision for the Worker EC2 Instance root disk."
     },
+    "WorkerEbsOptimized": {
+      "Type": "String",
+      "Description": "Enable optimized EBS I/O",
+      "Default": "true"
```
Comment (bcwaldon) [inline roles/configure/files/stack-template.json]: Let's not set a default here if we're going to set it in create-stack.sh

**Comment by bcwaldon** [comment]
@chadcatlett a couple of nits then LGTM

**Comment by chadcatlett** [comment]
@bcwaldon sorry for the delay, i've removed the defaults and updated the worker_instance_type commentary.

**Comment by bcwaldon** [comment]
@chadcatlett ok, I think we went one step too far. Now worker_instance_ebs_optimized isn't actually being set in the default case. Could you uncomment the field in the example config?

**Comment by chadcatlett** [comment]
@bcwaldon fixed, but don't merge. I'm going to throw in configurability for all the other instances. Sigh.

**Comment by bcwaldon** [comment]
standing by

**Comment by chadcatlett** [comment]
@bcwaldon - i've pushed my latest changes.

**Comment by jk0** [comment]
bump

**Comment by bcwaldon** [comment]
looks good!

**Pull request merged**